### DISCUSSION TITLE
How to install CVAT in AZURE Devops

### DISCUSSION THREAD

**Discussion by tejan74** [cvat-ai/cvat#4358] [Q&A]
Hi @azhavoro

I am going to deploy CVAT in Azure devops. Please provide any documentation related to CVAT deployment in Azure or kubenetes yamls  for latest functions.

**Comment by Gelesh** [comment]
You can install it on a Docket on a VM , or may I know what other kind of Azure service you are thinking about

**Comment by Mattral** [comment]
Here's a general approach to deploying CVAT on Azure DevOps and Kubernetes:

Azure DevOps:

Azure VM Deployment: Start by deploying one or more virtual machines (VMs) on Azure where you intend to host CVAT. Ensure that the VMs meet the hardware and software requirements for CVAT.

Configure Networking: Set up networking to allow access to the VMs, including configuring network security groups (NSGs) and opening the necessary ports for HTTP/HTTPS traffic.

Install Docker: Install Docker on your Azure VMs to prepare them for containerized application deployment.

Deploy CVAT: Build a Docker image of CVAT and push it to a container registry (e.g., Azure Container Registry). Then, deploy CVAT containers on your Azure VMs using tools like Docker Compose or Kubernetes.

Kubernetes:

Set Up Kubernetes Cluster: Create or use an existing Kubernetes cluster on your preferred platform (e.g., Azure Kubernetes Service - AKS, Google Kubernetes Engine - GKE).

Create Kubernetes YAMLs: Write Kubernetes YAML configuration files to define the necessary resources for CVAT deployment, including deployments, services, ingress controllers, and persistent volumes. You may need to configure environment-specific variables, such as database connections and storage.

Deploy CVAT: Apply the Kubernetes YAMLs to your cluster using kubectl apply -f filename.yaml. This will create and manage the CVAT pods, services, and associated resources.

Ingress Configuration: Configure the Kubernetes Ingress resource to route external traffic to your CVAT service.

Scaling and Monitoring: Depending on your requirements, configure autoscaling for CVAT pods and set up monitoring and logging solutions for your Kubernetes cluster.

To get the latest Kubernetes YAMLs and deployment instructions for CVAT, I recommend checking the official CVAT GitHub repository, community forums, or blogs. The CVAT community may have developed specific Kubernetes deployment files or scripts for the latest CVAT versions.

Additionally, Azure DevOps offers integration and deployment pipelines, so you might find Azure DevOps-specific deployment pipelines for CVAT in Azure DevOps documentation or community resources.
